internal/provider/scim_group_datasource.go

```diff
@@ -0,0 +1,143 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/temporalio/terraform-provider-temporalcloud/internal/client"
+	cloudservicev1 "go.temporal.io/cloud-sdk/api/cloudservice/v1"
+	"go.temporal.io/cloud-sdk/api/identity/v1"
+)
+
+var (
+	_ datasource.DataSource              = &scimGroupDataSource{}
+	_ datasource.DataSourceWithConfigure = &scimGroupDataSource{}
+)
+
+func NewSCIMGroupDataSource() datasource.DataSource {
+	return &scimGroupDataSource{}
+}
+
+type scimGroupDataModel struct {
+	ID    types.String `tfsdk:"id"`
+	Name  types.String `tfsdk:"name"`
+	IdpId types.String `tfsdk:"idp_id"`
+}
+
+type scimGroupDataSource struct {
+	client *client.Client
+}
+
+func (d *scimGroupDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *scimGroupDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_scim_group"
```

Comment:
just curious why we are appending this string to the type and it's not a part of the type itself?

Reply:
This follows existing patterns, but I'm guessing it's because all of them would have the same prefix, and this allows us to change that if we needed to?